pkg/sfu/sendsidebwe/packet_tracker.go

```diff
@@ -4,49 +4,94 @@ import (
 	"sync"
 	"time"
 
+	"github.com/MicahParks/peakdetect"
+	"github.com/frostbyte73/core"
+	"github.com/gammazero/deque"
 	"github.com/livekit/protocol/logger"
+	"github.com/pion/rtcp"
 )
 
 type packetInfo struct {
-	sendTime    time.Time
+	sendTime time.Time
+	// SSBWE-TODO: may need a feedback report time to detect stale reports
 	arrivalTime int64
 	headerSize  uint16
 	payloadSize uint16
+	// SSBWE-TODO: possibly add the following fields - pertaining to this packet,
+	// idea is to be able to traverse back and find last packet with clean signal(s)
+	// in order to figure out bitrate at which congestion triggered.
+	//    MPETau - Mean Percentage Error trend - potentially an early signal of impending congestion
+	//    PeakDetectorSignal - A sustained peak could indicate definite congestion
+	//    AcknowledgedBitrate - Bitrate acknowledged by feedback
+	//    ProbePacketInfo - When probing, these packets could be analyzed separately to check if probing is successful
```

Comment:
All these are things to be explored. The idea is to record these at packet level (potentially this can be done for all packets, but that could become expensive, so potentially a sub-sampled version).

The things I am trying to explore are
1. Have two analysers run to get two different kind of signals. One is fast acting and can give a coarse signal of early indication of impending congestion. This can be used to pause any layer scale up. From that point on, things could go either way (towards congestion or back to normal in which case the layer scale up pause can be removed). The second signal is more reliable and can confirm congestion.
2. Am trying Mean Percentage Error (https://en.wikipedia.org/wiki/Mean_percentage_error) trend for coarse signal and peak detector mentioned above for the stronger signal. Not convinced either is the right choice. Need to think/dig more. But, committing this as a concept of the two-pronged approach thought process to get more feedback.
3. The acknowledged bitrate (or measured bitrate) at different instances (i. e. corresponding to when that packet was received at the remote side). The idea here is to hunt for an estimate when there is congestion. When congestion hits, we can walk back this list and find a point when there was no congestion and pick out the bitrate at that time as the bit rate which started the congestion and hence can be used as the current channel capacity (may not be exactly that value, but some value around that number).
4. When probing for more bit rate, want to see if we can mark those packets and analyse them in isolation. Is it possible to probe for something like 100 ms (or even shorter window) at some desired rate and check if channel can sustain that rate?